intake_esm/core.py

```diff
@@ -157,11 +169,20 @@ def serialize(self, name, directory=None):
             json_file_name = directory / json_file_name
 
         collection_data = self._col_data.copy()
-        collection_data['catalog_file'] = csv_file_name.as_posix()
+        collection_data = _clear_old_catalog(collection_data)
         collection_data['id'] = name
 
-        print(f'Writing csv catalog to: {csv_file_name}')
-        self.df.to_csv(csv_file_name, compression='gzip', index=False)
+        if catalog_type == 'file':
+            collection_data['catalog_file'] = csv_file_name.as_posix()
+            print(f'Writing csv catalog to: {csv_file_name}')
+            self.df.to_csv(csv_file_name, compression='gzip', index=False)
+        else:
+            catalog_length = len(self.df)
+            print(f'Writing catalog with {catalog_length} entries into: {json_file_name}')
+            if catalog_length > 100:
+                print("  (call with catalog_type='file' to save catalog as a separate CSV file)")
```

Comment:
Thinking more about the `100` threshold here,  I am now in favor of removing this suggestion. IMHO, I don't think that there's a better default here. So, instead of us guessing what this default is, we should let user decide what do. It might be more useful to move this suggestion to the docstring instead.  After the parameter section in the docstring, we can add a `Notes` section:

```python
""""
Notes
------
# Add suggestion here
"""
```

Reply:
@andersy005  I'm not sure if I should add a callout to every message I make here, but anyway...I have pushed the change you suggested.

Reply:
> I'm not sure if I should add a callout to every message I make here

You don't have to :) I get notified by GitHub.